src/dune_rules/dune_file.ml

```diff
@@ -539,20 +539,11 @@ module Library = struct
       else
         (* Old behavior: if old parser succeeds, return that. Otherwise, if
            parsing the ordered set language succeeds, ask the user to upgrade to
-           a supported version. Otherwise, fail with the first error. *)
-        try_
-          (Mode_conf.Lib.Set.decode >>| fun modes -> `Modes modes)
-          (fun exn ->
-            try_
-              ( Mode_conf.Lib.Set.decode_osl ~stanza_loc project >>| fun modes ->
-                if dune_version >= expected_version then `Modes modes
-                else `Upgrade )
-              (fun _ -> raise exn))
-        >>| function
-        | `Modes modes -> modes
-        | `Upgrade ->
-          Syntax.Error.since stanza_loc Stanza.syntax expected_version
-            ~what:"Ordered set language for modes"
+           a supported version *)
+        try_ Mode_conf.Lib.Set.decode (fun _exn ->
```

Comment:
Can the `Mode_conf.Lib.Set.decode` expression ever raise?

Reply:
It can raise if there’s an ordered set language in the field

Reply:
But as the param passed to `try`, `Mode_conf.Lib.Set.decode` is not being applied to anything. How can the evaluated expression raise if it's not applied to anything? I read this code the same as:

```ml
let raise _s = failwith "foo"

let _ = try raise with | _exn -> print_endline
```

There's no point on wrapping with `try` in the example above.

Reply:
Because it’s inside the parser monad

Reply:
Thanks missed that 🤦